lib/examples/did-test-suite.rs

```diff
@@ -0,0 +1,629 @@
+    serde_json::to_writer_pretty(writer, &report).unwrap();
+}
+
+async fn report_dereferencer_web() {
+    let mut report = DIDResolverImplementation {
+        did_method: "did:web".to_string(),
+        implementation: "https://github.com/spruceid/ssi/tree/main/did-web".to_string(),
+        implementer: "Spruce Systems, Inc.".to_string(),
+        expected_outcomes: HashMap::new(),
+        executions: Vec::new(),
+    };
+
+    for did_url in vec!["did:web:did.actor:nonexistent"] {
+        report
+            .dereference(
+                &did_web::DIDWeb,
+                did_url,
+                &DereferencingInputMetadata::default(),
+            )
+            .await;
+    }
+
+    let writer = std::io::BufWriter::new(std::io::stdout());
+    serde_json::to_writer_pretty(writer, &report).unwrap();
+}
+
+async fn report_dereferencer_tz() {
+    let did_tz = did_tz::DIDTz::default();
+    let mut report = DIDResolverImplementation {
+        did_method: "did:tz".to_string(),
+        implementation: "https://github.com/spruceid/ssi/tree/main/did-tezos".to_string(),
+        implementer: "Spruce Systems, Inc.".to_string(),
+        expected_outcomes: HashMap::new(),
+        executions: Vec::new(),
+    };
+    for did_url in vec![
+        "did:tz:tz1YwA1FwpgLtc1G8DKbbZ6e6PTb1dQMRn5x",
+        "did:tz:tz1YwA1FwpgLtc1G8DKbbZ6e6PTb1dQMRn5x#blockchainAccountId",
+    ] {
+        report
+            .dereference(&did_tz, did_url, &DereferencingInputMetadata::default())
+            .await;
+    }
+
+    let writer = std::io::BufWriter::new(std::io::stdout());
+    serde_json::to_writer_pretty(writer, &report).unwrap();
+}
+
+async fn report_method(mut args: Args) {
+    let method = args.next().expect("expected method argument");
+    args.next().ok_or(()).expect_err("unexpected argument");
+    match &method[..] {
+        "key" => report_method_key().await,
+        "web" => report_method_web().await,
+        "tz" => report_method_tz().await,
+        method => panic!("unknown method {}", method),
+    }
+}
+
+async fn report_resolver(mut args: Args) {
+    let method = args.next().expect("expected method argument");
+    args.next().ok_or(()).expect_err("unexpected argument");
+    match &method[..] {
+        "key" => report_resolver_key().await,
+        "web" => report_resolver_web().await,
+        "tz" => report_resolver_tz().await,
+        method => panic!("unknown method {}", method),
+    }
+}
+
+async fn report_dereferencer(mut args: Args) {
+    let method = args.next().expect("expected method argument");
+    args.next().ok_or(()).expect_err("unexpected argument");
+    match &method[..] {
+        "key" => report_dereferencer_key().await,
+        "web" => report_dereferencer_web().await,
+        "tz" => report_dereferencer_tz().await,
+        method => panic!("unknown method {}", method),
+    }
+}
+
+#[tokio::main]
+async fn main() {
+    let mut args = std::env::args();
```

Comment:
Don't want to just use structopt?

Reply:
Probably should... but felt that this would be quicker